src/pages/WorkloadDetails/WorkloadDetailsPage.tsx

```diff
@@ -53,17 +38,48 @@ class WorkloadDetails extends React.Component<RouteComponentProps<WorkloadId>, W
     return '/namespaces/' + this.props.match.params.namespace + '/workloads/' + this.props.match.params.workload;
   }
 
+  // All information for validations is fetched in the workload, no need to add another call
+  workloadValidations(workload: Workload): Validations {
+    const noIstiosidecar: ObjectCheck = { message: 'Pod has no Istio sidecar', severity: 'warning', path: '' };
+    const noAppLabel: ObjectCheck = { message: 'Pod has no app label', severity: 'warning', path: '' };
+    const noVersionLabel: ObjectCheck = { message: 'Pod has no version label', severity: 'warning', path: '' };
+
+    const validations: Validations = {};
+    if (workload.pods.length > 0) {
+      validations['pod'] = {};
+      workload.pods.forEach(pod => {
+        validations['pod'][pod.name] = {
+          name: pod.name,
+          objectType: 'pod',
+          valid: true,
+          checks: []
+        };
+        if (!pod.istioContainers || pod.istioContainers.length === 0) {
+          validations['pod'][pod.name].checks.push(noIstiosidecar);
+        }
+        if (!pod.labels) {
+          validations['pod'][pod.name].checks.push(noAppLabel);
+          validations['pod'][pod.name].checks.push(noVersionLabel);
+        } else {
+          if (!('app' in pod.labels)) {
```

Comment:
these label names 'app' and 'version' should not be hardcoded
They could be passed in the workloads results, or another possibility would be to perform labels validation from the backend only.

Reply:
+1
Pod validation can be done from the model.Pod as it is created.
I willl address that.
Thanks for checking it.